common/OpTestHMC.py

```diff
@@ -0,0 +1,348 @@
+            console.sendline("echo $?")
+            console.send('\r')
+            # Time to accumulate both outputs
+            time.sleep(STALLTIME)
+            rc = console.expect(["\[console-pexpect\]#$"], timeout=timeout)
+            if rc != 0:
+                exitcode = int(console.before)
+                log.debug("# LAST COMMAND EXIT CODE %d (%s)" % (exitcode, repr(console.before)))
+        except pexpect.TIMEOUT as e:
+            log.debug(e)
+            log.debug("# TIMEOUT waiting for command to finish.")
+            log.debug("# Attempting to control-c")
+            try:
+                console.sendcontrol('c')
+                rc = console.expect(["\[console-pexpect\]#$"], 10)
+                if rc == 0:
+                    raise CommandFailed(cmd, "TIMEOUT", -1)
+            except pexpect.TIMEOUT:
+                log.error("# Timeout trying to kill timed-out command.")
+                log.error("# Failing current command and attempting to continue")
+                self.deactivate_lpar_console()
+                raise CommandFailed("console", "timeout", -1)
+            raise e
+
+        if rc == 0:
+            res = console.before
+            res = res.split(cmd)
+            return res[-1].splitlines()[1:-2]
+        else:
+            res = output
+            res = res.splitlines()
+            if exitcode != 0:
+                raise CommandFailed(cmd, res, exitcode)
+            return res
+
+    def get_console(self):
+        console = self.get_console_prompt()
+        time.sleep(STALLTIME)
+        l_rc = console.expect(["login:", pexpect.TIMEOUT], timeout=WAITTIME)
+        if l_rc == 0:
+            console.send('\r')
+            console = self.wait_login_prompt(console, self.lpar_user, self.lpar_password)
+        else:
+            time.sleep(STALLTIME)
+            console.send('\r')
+            console.sendline('PS1=[console-pexpect]\#')
+            console.send('\r')
+            time.sleep(STALLTIME)
+            l_rc = console.expect(["\[console-pexpect\]#$", pexpect.TIMEOUT], timeout=WAITTIME)
+            if l_rc == 0:
+                log.debug("Shell prompt changed")
+            else:
+                console.send('\r')
+                log.debug("Waiting till booting!")
+                console = self.wait_login_prompt(console, self.lpar_user, self.lpar_password)
+        return console
+
+    def check_vterm(self):
+        return self.vterm
+
+    def wait_login_prompt(self, console, username=None, password=None):
+        if not console:
+            raise OpTestError("Console is not provided")
+        # Assuming 'Normal' boot set in LPAR profile
+        # We wait for upto 500 seconds for LPAR to boot to OS
+        console.buffer = ""
+        console.send('\r')
+        time.sleep(STALLTIME)
+        log.debug("Waiting for login screen")
+        i = console.expect(
+                ["login:", "\[console-pexpect\]#$", pexpect.TIMEOUT], timeout=BOOTTIME)
```

Comment:
Suggest using the utility to build the prompt to provide a system wide common abstraction.